core/vm/contracts.go

```diff
@@ -358,3 +375,49 @@ func (c *bn256Pairing) Run(input []byte) ([]byte, error) {
 	}
 	return false32Byte, nil
 }
+
+type blake2F struct{}
+
+func (c *blake2F) RequiredGas(input []byte) uint64 {
+	return 1 //TODO: implement
+}
+
+func (c *blake2F) Run(input []byte) ([]byte, error) {
+	var h [8]uint64
+	h[0] = binary.BigEndian.Uint64(input[0:8])
+	h[1] = binary.BigEndian.Uint64(input[8:16])
+	h[2] = binary.BigEndian.Uint64(input[16:24])
+	h[3] = binary.BigEndian.Uint64(input[24:32])
+	h[4] = binary.BigEndian.Uint64(input[32:40])
+	h[5] = binary.BigEndian.Uint64(input[40:48])
+	h[6] = binary.BigEndian.Uint64(input[48:56])
+	h[7] = binary.BigEndian.Uint64(input[56:64])
+
+	var m [blake2.BlockSize]byte
+	copy(m[:], input[64:192])
+
+	var t [2]uint64
+	t[0] = binary.BigEndian.Uint64(input[216:224])
+	t[1] = binary.BigEndian.Uint64(input[248:256])
+
+	var f bool
+	if input[287] == 1 {
+		f = true
+	}
+
+	rounds := binary.BigEndian.Uint32(input[316:320])
```

Comment:
actually, `panic` on shorter than  `320` bytes in length

Reply:
Yep, it was wrong. We now expect the input to be exactly 213 bytes long.